scylla-cql/src/frame/response/type_parser.rs

```diff
@@ -0,0 +1,316 @@
+            "DurationType" => Ok(ColumnType::Native(NativeType::Duration)),
+            "FloatType" => Ok(ColumnType::Native(NativeType::Float)),
+            "InetAddressType" => Ok(ColumnType::Native(NativeType::Inet)),
+            "Int32Type" => Ok(ColumnType::Native(NativeType::Int)),
+            "IntegerType" => Ok(ColumnType::Native(NativeType::Varint)),
+            "LongType" => Ok(ColumnType::Native(NativeType::BigInt)),
+            "SimpleDateType" => Ok(ColumnType::Native(NativeType::Date)),
+            "ShortType" => Ok(ColumnType::Native(NativeType::SmallInt)),
+            "UTF8Type" => Ok(ColumnType::Native(NativeType::Text)),
+            "ByteType" => Ok(ColumnType::Native(NativeType::TinyInt)),
+            "UUIDType" => Ok(ColumnType::Native(NativeType::Uuid)),
+            "TimeUUIDType" => Ok(ColumnType::Native(NativeType::Timeuuid)),
+            "SmallIntType" => Ok(ColumnType::Native(NativeType::SmallInt)),
+            "TinyIntType" => Ok(ColumnType::Native(NativeType::TinyInt)),
+            "TimeType" => Ok(ColumnType::Native(NativeType::Time)),
+            "TimestampType" => Ok(ColumnType::Native(NativeType::Timestamp)),
+            _ => Err(CustomTypeParseError::UnknownSimpleCustomTypeName(
+                name.into(),
+            )),
+        }
+    }
+
+    fn get_type_parameters(
+        &mut self,
+    ) -> Result<
+        impl Iterator<Item = Result<ColumnType<'result>, CustomTypeParseError>> + '_,
+        CustomTypeParseError,
+    > {
+        if self.parser.is_at_eof() {
+            return Ok(Either::Left(std::iter::empty()));
+        }
+        self.parser = self
+            .parser
+            .accept("(")
+            .map_err(|_| CustomTypeParseError::BadCharacter)?;
+
+        Ok(Either::Right(std::iter::from_fn(|| {
+            self.skip_blank_and_comma();
+            if self.parser.is_at_eof() {
+                return Some(Err(CustomTypeParseError::UnexpectedEndOfInput));
+            }
+            let result = self.parser.accept(")");
+            match result {
+                Ok(parser) => {
+                    self.parser = parser;
+                    None
+                }
+                Err(_) => Some(self.do_parse()),
+            }
+        })))
+    }
+
+    /// Parse a string of hexadecimal representation of bytes into a byte vector.
+    fn from_hex(s: &'result str) -> Result<Vec<u8>, CustomTypeParseError> {
+        if s.len() % 2 != 0 {
+            return Err(CustomTypeParseError::BadHexString);
+        }
+        for c in s.chars() {
+            if !c.is_ascii_hexdigit() {
+                return Err(CustomTypeParseError::BadHexString);
+            }
+        }
+        let mut bytes = Vec::new();
+        for i in 0..s.len() / 2 {
+            let byte = u8::from_str_radix(&s[i * 2..i * 2 + 2], 16)
+                .map_err(|_| CustomTypeParseError::BadHexString)?;
+            bytes.push(byte);
+        }
+        Ok(bytes)
```

Comment:
I dislike that bracket indexing, but I don't have a good idea on how to change it. @wprzytula maybe you'll think of something?

Reply:
I don't really see an advantage. Both versions contain a panic (that is verified to be impossible to trigger).

Reply:
The advantage is in readability. It's much easier to parse (at least for me) such FP code, especially that it uses clear wording, than to parse those slice indexing arithmetics.

Reply:
Ok